### ISSUE TITLE
Deployment method

### ISSUE THREAD

**Issue by g0ulash** [MKaptein/streamingbandit#3]
MKaptein: "How can we install streampy (including Redis, Mongo, etc.) on a machine in 1 go?"

**Comment by VincentGijsen** [comment]
easy (the ugly way),

just create a bash-script, save it to the root and call it `install_in_one_go.sh`
then execute it `./install_in_one_go.sh` (assumes brew is installed, on MAC OSX), you may want to add some if conditions for Linux users to replace brew with `apt-get` and a `wget` to pull the latest mongodb dpkg package

e.g.

``` bash
#!/bin/bash
brew install mongodb
brew install redis
cd app
python setup.py install
```

**Comment by g0ulash** [comment]
Yes, this is one way to do it and something that I would prefer (we could also provide a complete image for the non-technical people). However, apt-getting redis does not always install it as a daemon/service and redis is outdated most of the times via that way. And the app right now does not yet provide an upstart/daemon script, which would be cool to include as well - but we could add a nice installation guide for that one. (For installing redis as a service, see: https://www.digitalocean.com/community/tutorials/how-to-install-and-use-redis)

**Comment by g0ulash** [comment]
Multiple ideas for the future:

1. Host at a dedicated location;
2. Make available as Amazon server image (needs continuous support/updating);
3. Make available as Docker container (also needs continuous support/updating);
4. Have a good description of manual installation (this is already somewhat included in the README.md).

**Comment by g0ulash** [comment]
Perhaps we may consider deploying with an NGINX installation/config file as well?

**Comment by g0ulash** [comment]
We should also supply, with all different deployment methods, a setup script that will make an admin user on first start-up. (I have such a script available, but not yet implemented in setup.py or anything else).

**Comment by MKaptein** [comment]
Just include it in setup.py; that solves the problem. That it will solve it for all deployment methods... 

(since these are:
1. Manually; follow the readme. We tell people to run setup.py
2. Docker; containers allow you to specify scripts that should run on startup
3. Our hosted online version; here this is not applicable
)

**Comment by g0ulash** [comment]
I've been reading into Docker, and it seems that to fully deploy our software, we need to use docker-compose. What this means is that there is no straightforward way to make 1 docker image of our software, put it on Dockerhub and let people run it like `docker run -p 8080:8080 streamingbandit/latest`. If we really want this to be easy, I'm not 100% sure if Docker is the way to go. An example of how to deploy StreamingBandit would be as explained here: https://nickjanetakis.com/blog/dockerize-a-flask-celery-and-redis-application-with-docker-compose and here: http://anandmanisankar.com/posts/docker-container-nginx-node-redis-example/

**Comment by MKaptein** [comment]
Hi,

As far as I understand we can do a simple docker, but only for the front-end and back-end separately (as is now described in the paper). The docker-compose would allow us to set both up at the same time in one easy go (at least, that’s my understanding of it). 

If you have an alternative suggestion I am happy to go with that, but we really need an easy way to get up and running. Also, I guess the docker compose is doable as well?

**Comment by g0ulash** [comment]
I started with a Dockerfile and docker-compose.yml in 3d67d90. There are two issues remaining:

1. A minor issue, but the current Dockerfile is based on python:3.5-jessie. This results in an image that is over 1GB large, only due to the fact that this python image is large. There are smaller images available, of which the best is python:3.5-alpine, but it's not straightforward to install e.g. scipy and numpy, because BLAS/Lapack is not installed in Alpine (since it's very very very barebone).

2. I haven't found a decent way to add an admin user when the MongoDB is first instantiated. It's also apparently impossible to run `pip install setup.py` and you're limited to `pip install -r requirements.txt`. Using a setup.py file would probably enable us to add an admin user to the MongoDB instance. 

To be continued...

**Comment by g0ulash** [comment]
Deployment method is finished. We only need to update the README.md and the documentation with a nice installation guide.
